### DISCUSSION TITLE
fermi hubbard model with complex tunneling

### DISCUSSION THREAD

**Discussion by Yuanningss** [QuSpin/QuSpin#527] [General]
Hi 

I am now considering fermi hubbard model with periodic driving, here is the full hamiltonian: 
<img width="336" alt="hamiltonian" src="https://user-images.githubusercontent.com/104153268/164646537-2d1a74b9-0810-4a95-a25f-c3160e951346.png">

However, when I implement this hamiltonian corresponding to this link https://github.com/weinbe58/QuSpin/blob/master/examples/notebooks/FHM.ipynb
and specifically as I write the complex tunneling in this fashion: 

dynamic =[
          ['-+|',hop_right, drivingphase2, drivingphase2_args],  # spin up\P\hops right
          ['+-|',hop_left, drivingphase1, drivingphase1_args],  # spin up\P\hops left
          ['|-+',hop_right, drivingphase2, drivingphase2_args],  # spin down\P\hops right
          ['|+-',hop_left, drivingphase1, drivingphase1_args],  # spin down\P\hops left
   
where : 

hop_right =[[-J_par_1,i,(i+2)%N] for i in range(1,N,2)] # PBC 
hop_left =[[J_par_1,i,(i+2)%N] for i in range(1,N,2)]  # PBC 

def drivingphase1(t):
    return np.exp(1j*sin(wt))
drivingphase1_args=[ ]

def drivingphase2(t):
    return np.exp(-1j*sin(wt))

I encountered this error : ComplexWarning: Casting complex values to real discards the imaginary part.
This mean that only the real part of the tunneling coefficient is considered. However, I need the imaginary part because when 
considering the periodic driving, a time dependent phase (called Peierls phase) is added

. 
There I am wondering how to fix this issue by allowing the tunneling coefficient be complex ? 

Thanks

**Comment by JiahaoYao** [comment]
Hi @Yuanningss , thanks for your interest in `Quspin`. This seems like the type error. You might be able to change 

from 
```python 
hamiltonian(static,dynamic,basis=basis,dtype=np.float64,**no_checks)
```

to 
```python 
hamiltonian(static,dynamic,basis=basis,dtype=np.complex128,**no_checks)
```

so that the `complex` type can be supported! Feel free to follow up on this!
